### PR TITLE
[snapshot] Sort screenshots.html by screen name

### PR DESCRIPTION
**Pull request by claesjacobsson**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

The screenshots.html page currently sorts the screenshots by language and device type. 
Sometimes it's useful to compare all localizations of a screen side by side. This PR makes that possible by providing two sorting options: _By Language_ and _By Screen_.

Fixes issue #13626.

### Description

_reports_generator.rb_ now produces two data objects (_data_by_language_ and _data_by_screen_) for _page.html.erb_ to use when creating _screenshots.html_.

_screenshots.html_ has been updated with a top menu bar to jump between the two sorting options.

As far as I understand, the current version only tests the device names being used in _reports_generator_. I have not made any changes and didn't see a reason to add any tests at this time.

### CODE REVIEW

**Review by janpio — APPROVED** [review]
Works as described, looks like this:

image

The only thing that one could nitpick here is that it only works with Javascript enabled. In my testing, with Javascript disabled in Firefox, I couldn't see any screenshots at all. 

Maybe the tabs could both be visible by default, and only set to `display:none` via Javascript? (Add another headline above the "tab content" so it is clear what is actually shown below, and then also hide that new headline if you want to) What do you think?

**Comment by claesjacobsson** [comment]
Didn't consider that. Your idea makes sense. Would you like me to make a separate PR for that or add to this one?

**Comment by janpio** [comment]
If you can, just commit it to this PR's branch. Then we can merge it in one swoop.

**Comment by claesjacobsson** [comment]
Done!

File: snapshot/lib/snapshot/page.html.erb [hunk 1]
```diff
@@ -64,23 +90,48 @@
...
-        </td><% end %>
-      </tr><% end %>
-    </table><% end %>
+  <body>
+    <div id="sortMenu">
+      <button id="defaultTab" class="tabLink" onclick="openTab(event, 'byLanguage')">By Language</button>
+      <button class="tabLink" onclick="openTab(event, 'byScreen')">By Screen</button>
+    </div>
+    <div id="byLanguage" class="tabContent"><h1 class="tabTitle">By Language:</h1><% image_counter = 0 %><% @data_by_language.each do |language, content| %>
+      <h1 id="<%= language %>"><%= language %></h1>
```
Comment (janpio) [inline snapshot/lib/snapshot/page.html.erb]: I know this doesn't really matter here, but could we bump those down to h2 so the semantics are more correct? Thanks.

Comment (claesjacobsson) [reply snapshot/lib/snapshot/page.html.erb]: For the language, you mean?

Comment (janpio) [reply snapshot/lib/snapshot/page.html.erb]: yep. but actually, I'm fine either way.

Comment (claesjacobsson) [reply snapshot/lib/snapshot/page.html.erb]: I can fix that. Makes sense.

**Review by janpio — APPROVED** [review]
Looks perfect now, even without JS:

image

And with JS:

image
and
image

**Comment by janpio** [comment]
Thanks for applying my suggestions to this PR @claesjacobsson, the result is awesome! 🚀 💯

**Pull request merged**